.github/workflows/ci/pkg/environment/environment.go

```diff
@@ -111,22 +118,94 @@ func New(c Config) (*PullRequestEnvironment, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	// Check if the pull request has changes to the `docs/` directory as that will affect who
+	// the required reviewers are.
+	docChanges, codeChanges, err := hasChanges(c.Context, pr, c.Client)
+	if err != nil {
+		// Log the error, don't fail the whole run because it couldn't detect if the pull request has docs.
+		log.Errorf("error while detecting pull request for docs changes: %v, skipping assigning docs reviewers", err)
+	}
+
 	return &PullRequestEnvironment{
 		Client:           c.Client,
 		reviewers:        c.Reviewers,
 		defaultReviewers: c.Reviewers[ci.AnyAuthor],
 		Metadata:         pr,
+		HasDocsChanges:   docChanges,
+		HasCodeChanges:   codeChanges,
 	}, nil
 }
 
+// hasChanges determines if the pull request has docs and/or code changes.
+func hasChanges(ctx context.Context, pr *Metadata, clt *github.Client) (hasDocsChanges bool, hasCodeChanges bool, err error) {
+	files, err := getPullRequestFiles(ctx, pr, clt)
+	if err != nil {
+		return false, true, trace.Wrap(err)
+	}
+	for _, file := range files {
+		if file.Filename == nil {
+			return false, true, trace.BadParameter("pull request file name is nil")
+		}
+		if hasDocChanges(*file.Filename) {
+			hasDocsChanges = true
+		} else {
+			hasCodeChanges = true
+		}
+
+	}
+	return hasDocsChanges, hasCodeChanges, nil
+}
+
+// hasDocChanges checks if the file name contains a prefix or suffix that would suggest a change to docs.
+// Prefix of "rfd/" or "docs/".
+// Suffix of ".mdx", or ".md".
+func hasDocChanges(filename string) bool {
+	if strings.HasPrefix(filename, ci.VendorPrefix) {
+		return false
+	}
+	return strings.HasPrefix(filename, ci.DocsPrefix) ||
+		strings.HasSuffix(filename, ci.MdSuffix) ||
+		strings.HasSuffix(filename, ci.MdxSuffix) ||
+		strings.HasPrefix(filename, ci.RfdPrefix)
```

Comment:
Maybe put this into a slice of prefixes similar to what you did with `RepoAdmins`?

Reply:
I would need 2 slices, one with prefixes and suffixes. I think this is a bit cleaner. If there are other conditions in the future that we need to add, maybe we can refactor then.